beacon-chain/slasher/receive.go

```diff
@@ -69,27 +61,33 @@ func (s *Service) processQueuedAttestations(ctx context.Context, epochTicker <-c
 	for {
 		select {
 		case currentEpoch := <-epochTicker:
-			s.attestationQueueLock.Lock()
-			attestations := s.attestationQueue
-			s.attestationQueue = make([]*slashertypes.IndexedAttestationWrapper, 0)
-			s.attestationQueueLock.Unlock()
+			// We take all the attestations in the queue and filter out
+			// those which are valid now and valid in the future.
+			attestations := s.attsQueue.dequeue()
+			validAtts, validInFuture, numDropped := s.validateAttestationIntegrity(attestations, currentEpoch)
```

Comment:
Would this mean we process the same future attestations validity multiple times?